xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-test/xwiki-platform-index-test-tests/src/test/it/org/xwiki/index/test/ui/DeletedAttachmentsTest.java

```diff
@@ -0,0 +1,89 @@
+ /*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.index.test.ui;
+
+import java.io.ByteArrayInputStream;
+import java.util.Calendar;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.xwiki.index.test.po.DeletedAttachmentsPage;
+import org.xwiki.test.ui.AbstractTest;
+import org.xwiki.test.ui.TestUtils;
+import org.xwiki.test.ui.po.AttachmentsPane;
+import org.xwiki.test.ui.po.LiveTableElement;
+import org.xwiki.test.ui.po.ViewPage;
+
+ /**
+  * Tests for the DeleteAttachments page.
+  *
+  * @since 12.2RC1
+  */
+ public class DeletedAttachmentsTest extends AbstractTest
+ {
+     @Before
+     public void setUp() throws Exception
+     {
+         getUtil().loginAsSuperAdmin();
+
+         if (!getUtil().pageExists(getTestClassName(), "TestAttachments")) {
```

Comment:
There should be no "'if" in a test. This means the tests is not performing the right setup. It should remove pages it needs. Imagine that the page exists but in a wrong state.